fix: nested arrays with a single element not parsed correctly

Yihao-G opened the pull request:
There appears to be a bug in this Gem that when there is an array with just one element which is also an array, it flattens the inner array.

For example, 
* JSON5: `[[1]]`
* Actual result: `[1]`
* Expected result: `[[1]]`

This PR fixes it by marking each `value` inside the `array_elements` subtree with `:element`. This normalizes each value's subtree to a `Hash` containing the key of `:element`. It then takes the value for each `:element` to form an array when transforming `array_elements`.

taichi-ishitani commented on lib/rb_json5/parser/array.rb:
```diff
@@ -21,7 +21,8 @@ class Parser
 
     transform_rule(array_elements: subtree(:elements)) do
-      elements.is_a?(Array) && elements || [elements]
+      normalized_elements = elements.is_a?(Array) ? elements : [elements]
```
If `elements` is a hash then its result is like `[[:a, 0], [:b, 1]]` so this behavior makes wrong result.
Please ignore the above comment.

Yihao-G commented:
Thanks @taichi-ishitani 

I updated the code and rebased it on top of your master branch. Let me know if you need anything else to be changed

For updating the supported Ruby version -
Unfortunately, we are on an older version of Ruby. Is it possible for you to release this fix before bumping the minimum supported version, please?

taichi-ishitani commented:
@Yihao-G ,

Thank you for updateing PR! I will it soon.

>  Is it possible for you to release this fix before bumping the minimum supported version, please?

OK, I underestood. I will create a patch release after merging this PR.
